### PR TITLE
Modified Execution Plan Viewer

### PR DESCRIPTION
**Pull request by Paarth002**
## Summary
Modified Execution Plan Viewer: Uplifted Execution plan viewer support for relational flow and added execution plan viewer support for flow control execution nodes. Added form mode for Sequence Execution Node, Function Parameters Validation Node, Allocation Execution Node, Constant Execution Node and Relational TDS Instantiation Node.

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query-builder/style/_query-builder.scss [hunk 1]
```diff
@@ -833,17 +833,292 @@
...
+
+  &__info {
+    margin-bottom: 0.5em;
+  }
+
+  &__item {
+    width: 100%;
+    border: 0.1rem solid var(--color-dark-grey-300);
+    border-radius: 0.3rem;
+    font-size: 2rem;
```
Comment (xannem) [inline packages/legend-query-builder/style/_query-builder.scss:1040]: Is there a reason for the larger font-size of the execution node? Maybe we can keep it like the size of `Execution Node` size for consistency 

image

File: packages/legend-query-builder/style/_query-builder.scss [hunk 2]
```diff
@@ -833,17 +833,292 @@
...
+    &__label {
+      color: var(--color-white);
+    }
+
+    &__btn {
+      height: 3rem;
+      width: 100%;
+      display: block;
+      text-align: left;
+      font-size: large;
```
Comment (xannem) [inline packages/legend-query-builder/style/_query-builder.scss:1057]: Is there a reason for the larger font-size of the execution node btn being `large`? Maybe we can keep it like the size of `Execution Node` size for consistency 

image

File: packages/legend-query-builder/style/_query-builder.scss [hunk 3]
```diff
@@ -833,17 +833,292 @@
...
+  &__info {
+    margin-bottom: 0.5em;
+  }
+
+  &__item {
+    width: 100%;
+    border: 0.1rem solid var(--color-dark-grey-300);
+    border-radius: 0.3rem;
+    font-size: 2rem;
+    margin: 10px;
```
Comment (xannem) [inline packages/legend-query-builder/style/_query-builder.scss:1041]: Do we need `margin 10px` here, maybe margin: 10px 0? It currently gives extra spacing to the node items that is not present in the result type table

image

File: packages/legend-query-builder/style/_query-builder.scss [hunk 4]
```diff
@@ -833,17 +833,292 @@
...
+    width: 100%;
+  }
+
+  &__info {
+    margin-bottom: 0.5em;
+    display: inline-block;
+
+    &__var-name {
+      display: inline-block;
+      text-indent: 5rem;
```
Comment (xannem) [inline packages/legend-query-builder/style/_query-builder.scss:907]: Are the text-indents for aligning the detail values to the same line? 
image
Is this preferable @akphi, I'm not sure if its necessary to add the text indent like this, if you wanted it structured in same alignment I think you could also use a table so its column widths may be responsive?

File: packages/legend-query-builder/src/components/execution-plan/ConstantExecutionNodeViewer.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      {value !== '' && (
+        <div className="query-builder__constant__container">
+          <div>
+            <PanelListItem className="query-builder__constant__container__item__label">
+              Constant Execution Node Details
+            </PanelListItem>
+            <PanelDivider />
+            <pre>
+              <div className="query-builder__constant__container__item">
+                <div>value : {value}</div>
```
Comment (xannem) [inline packages/legend-query-builder/src/components/execution-plan/ConstantExecutionNodeViewer.tsx:62]:
```suggestion
                <div>value: {value}</div>
```

File: packages/legend-query-builder/src/components/execution-plan/ResultTypeViewer.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+export const ResultTypeViewer: React.FC<{
+  resultType: ResultType;
+}> = observer((props) => {
+  const { resultType } = props;
+  if (resultType instanceof DataTypeResultType) {
+    return <DataTypeResultTypeViewer resultType={resultType} />;
+  } else if (resultType instanceof TDSResultType) {
+    return <TDSResultTypeViewer resultType={resultType} />;
+  } else {
+    return <></>;
```
Comment (MauricioUyaguari) [inline packages/legend-query-builder/src/components/execution-plan/ResultTypeViewer.tsx:35]: return null

File: packages/legend-query-builder/src/components/execution-plan/FunctionParametersValidationNodeViewer.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+export const FunctionParametersValidationNodeViewer: React.FC<{
+  functionParameters: VariableExpression[];
+  parameterValidationContext: ParameterValidationContext[];
+  resultType: ResultType;
+  executionPlanState: ExecutionPlanState;
+}> = observer((props) => {
+  const { functionParameters, resultType, executionPlanState } = props;
+  const applicationStore = executionPlanState.applicationStore;
+
+  const showMultiplicity = (multiplicity: Multiplicity): string => {
```
Comment (MauricioUyaguari) [inline packages/legend-query-builder/src/components/execution-plan/FunctionParametersValidationNodeViewer.tsx:47]: pretty sure we have a helper method to render multiplicity already

**Review by MauricioUyaguari — APPROVED** [review]

**Comment by MauricioUyaguari** [comment]
lets address css comments and not repeating code but will merge to move along

**Pull request merged**